experimental/uefi/loader/src/server.rs

```diff
@@ -28,6 +28,19 @@ use proto::{
     UnaryRequest, UnaryResponse,
 };
 
+fn serialize_request(request: UnaryRequest) -> Vec<u8> {
+    // The payload is the request's body prepended with the 8 byte session_id.
+    // This takes adavantage of the session_id's fixed size to avoid needing
+    // to use a key/value pair binary serialization protocol.
```

Comment:
It would be good to perhaps list what are the factors that we should consider in the first place, since for instance IIRC we already have a protobuf definition of this struct, but we are not using that directly in order to avoid requiring a hard dependency on protobuf for the trusted UEFI app (@jul-sh please correct me if I'm wrong).

Reply:
It basically goes back to the previous discussion here: https://github.com/project-oak/oak/pull/2759#discussion_r856430066. We already have a protobuf definition of this struct, but adding protobuf to the UEFI app just for that seems overkill.

The important thing imo is that this structure is very simple. All the interesting bits are in the binary-serialized body, the only thing that isn't is the fixed size session_id. It's seems rather unlikely that we'd want to add any other key/value pairs to this struct. 

Imo how individual client requests are serialised also isn't necessarily linked to our framing layer (currently CBOR). The framing layer is about separating different messages of various types that are sent UEFI runtime (lookup data, control frames, user requests, etc pp). This seems like it's a layer underneath that.

Reply:
So, from the client point of view we have gRPC / protobuf, since we know that's already something that most clients will have to deal with anyways. Actually that may be transcoded to JSON + HTTP anyways, so perhaps even clients that don't want to depend on gRPC / protobus may not have to depend on it.

Instead between Untrusted and Trusted runtime, the requirements come mostly from the trusted runtime itself and its dependencies (since they all end up being part of the TCB); in addition, even though it is not a short term goal, I think we should think about other trusted runtimes being written in other languages in the future (C++ possibly for some other potential collaboration). In any case, I think that the reference format of this serialization layer should just be specified in a text file somewhere, similar to how like TCP packets are defined (and the spec should be sufficient to implement de/serializers in any language); e.g. https://en.wikipedia.org/wiki/Transmission_Control_Protocol#TCP_segment_structure . In our case, I think having a "spec" that says that the first 8 bytes are the session id and the rest are the payload seems enough to me, we can naively implement that in any language without any additional dependencies; we don't need backwards compatibility or other features that protobuf and CBOR are usually used for.